errbot/backends/irc.py

```diff
@@ -416,6 +430,21 @@ def on_dcc_connect(self, dcc, event):
     def on_dcc_disconnect(self, dcc, event):
         self.transfers.pop(dcc)
 
+    def on_endofnames(self, c, e):
+        # The e.arguments[0] contains the channel name.
+        # We filter that to avoid a misfire of the event.
+        room_name = e.arguments[0]
+        if room_name in self.__recently_joined_to:
```

Comment:
That seems like a very bad thing to do here. There's no guarantee that `on_join` (which puts the room in `_recently_joined_to`) is going to be handled by the same thread as the `on_endofnames` here.

And even if that were the case now, there's no guarantee it would remain that way in the future. If it were thread local and different threads handled it, it would cause a very subtle bug that would likely be extremely hard to track down.

I wouldn't worry too much about thread locking. Especially not in this instance, as room joins are pretty rare.